src/controllers/processors/SessionCleanupProcessor.ts

```diff
@@ -0,0 +1,41 @@
+import { Session } from 'ch-node-session-handler';
+import { Request } from 'express';
+import { provide } from 'inversify-binding-decorators';
+
+import { FormActionProcessor } from 'app/controllers/processors/FormActionProcessor';
+import { loggerInstance } from 'app/middleware/Logger';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { APPLICATION_DATA_UNDEFINED, SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+import { CONFIRMATION_PAGE_URI } from 'app/utils/Paths';
+
+@provide(SessionCleanupProcessor)
+export class SessionCleanupProcessor implements FormActionProcessor {
+
+    public async process(req: Request): Promise<void> {
+
+        const session: Session | undefined = req.session;
+
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const appData: ApplicationData | undefined = session.getExtraData<ApplicationData>(APPLICATION_DATA_KEY);
+
+        if (!appData) {
+            throw APPLICATION_DATA_UNDEFINED;
+        }
+
+        appData.submittedAppeal = appData.appeal;
+
+        delete appData.appeal;
+
+        appData.navigation.permissions = [CONFIRMATION_PAGE_URI];
```

Comment:
I understand why you are doing this, but what if a user submits an appeal, and then starts a new one, doesn't this mean they will have permissions to jump to the confirmation page at any time even though they might not have completed the new journey

Reply:
Yes they will be able to see the appeal they just submitted. Sounds like a feature to me.

Reply:
Harmless feature

Reply:
True, I guess you are using the appealSubmitted data for that